File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> rightPro(n);
        
        rightPro[n - 1] = 1;
        for(int i=n-2; i>=0; i--){
            rightPro[i] = rightPro[i+1] * nums[i+1];
        }
        
        int leftPro = 1;
        for(int i=0; i<n; i++){
            rightPro[i] = leftPro * rightPro[i];
            leftPro = leftPro * nums[i];
        }
        return rightPro;
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    const len = nums.length;
    const right = [];
    
    right[len-1] = 1;
    for(let i=len-2; i>=0; i--){
       right[i] = right[i+1] * nums[i+1];
    }
    
    let left = 1;
    for(const idx in nums){
        right[idx] = right[idx] * left;
        left = left * nums[idx];
    }
    return right;
};